plugins/lookup/merge_variables.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Thales Netherlands
+# Copyright (c) 2021, Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author:
+      - Roy Lenferink (@rlenferink)
+      - Mark Ettema (@m-a-r-k-e)
+    name: merge_variables
+    short_description: merge variables with a certain suffix
+    description:
+        - This lookup returns the merged result of all variables in scope that matches the given suffix optionally
+          starting with an initial value.
+    version_added: 6.1.0
+    options:
+      _terms:
+        description:
+          - The suffix of the variable name (all available variables will be checked with endswith(suffix))
+        required: true
+      pattern_type:
+        description:
+          - Change the way of searching for the specified pattern.
+        type: str
+        default: 'regex'
+        choices:
+          - prefix
+          - suffix
+          - regex
+        env:
+          - name: ANSIBLE_MERGE_VARIABLES_PATTERN_TYPE
+      initial_value:
+        description:
+          - An initial value to start with
+        required: false
+      override:
+        description:
+          - Return an error, print a warning or ignore it when a key will be overwritten.
+          - The default behavior C(error) makes the plugin fail when a key would be overwritten.
+          - When C(warn) and C(ignore) are used, note that it is important to know that the variables
+            are sorted by name before being merged. Keys for later variables in this order will overwrite
+            keys of the same name for variables earlier in this order. To avoid potential confusion,
+            better use I(override=error) whenever possible.
+        type: str
+        default: 'error'
+        choices:
+          - error
+          - warn
+          - ignore
+        env:
+          - name: ANSIBLE_MERGE_VARIABLES_OVERRIDE
```

Comment:
You could additionally allow to configure this (and the other options that can be configured with env variables) with an Ansible variable.

Reply:
I can't really find something on how to do this. Do you know where this is documented?

I also don't really need to set it via an Ansible variable, what's you opinion here? I would also be okay with adding this at a later stadium. It is easier to extend with options than to remove options, since that would be a breaking change.

Reply:
> I can't really find something on how to do this. Do you know where this is documented?

I'm not sure where it is documented, but there are numerous examples in this collection, for example: https://github.com/ansible-collections/community.general/blob/main/plugins/lookup/passwordstore.py#L121-L125 This plugin option also can be configured in ansible.cfg (`ini:`).

> I also don't really need to set it via an Ansible variable, what's you opinion here? I would also be okay with adding this at a later stadium. It is easier to extend with options than to remove options, since that would be a breaking change.

Some folks prefer environment variables (you apparently do), others prefer Ansible variables (like me). Offering both is usually best so everyone can use whatever they prefer.

Reply:
> Some folks prefer environment variables (you apparently do), others prefer Ansible variables (like me). Offering both is usually best so everyone can use whatever they prefer.

Looking at the `passwordstore.py` example, this wasn't too hard to configure. I just added the options.